src/Controllers/CMSExternalLinksController.php

```diff
@@ -24,22 +26,26 @@ class CMSExternalLinksController extends Controller
      */
     public function getJobStatus()
     {
-        // Set headers
-        HTTPCacheControlMiddleware::singleton()->setMaxAge(0);
-        $this->response
-            ->addHeader('Content-Type', 'application/json')
-            ->addHeader('Content-Encoding', 'UTF-8')
-            ->addHeader('X-Content-Type-Options', 'nosniff');
+        if (Permission::checkMember(Security::getCurrentUser(), ['CMS_ACCESS_CMSMain'])) {
```

Comment:
```suggestion
        if (!Permission::checkMember(Security::getCurrentUser(), ['CMS_ACCESS_CMSMain'])) {
            return $this->httpError(403, 'You do not have permission to access this resource');
        }
```
Should return early so don't need to indent everything